### PR TITLE
Add acquiring ACR access token using Azure.Identity and allow unauthenticated ACR endpoints 

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

- Acquire ACR access token using Azure.Identity
- To acquire ACR access token we go through the following steps
  - Check if a credential is provided when registering the repository, if it then use it.
  - If no credential provided at registration then
    - Check if the ACR endpoint can be accessed without a token
    - If not, try using Azure.Identity to get the az access token, then ACR refresh token and then ACR access token.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Review by alerickson — APPROVED** [review]

**Comment by alerickson** [comment]
resolves #1580

File: .ci/test.yml [hunk 1]
```diff
@@ -78,6 +80,35 @@ jobs:
...
+    displayName: 'Set UsingAzAuth environment variable'
+    condition: eq(${{ parameters.useAzAuth }}, true)
+
+  - task: AzurePowerShell@5
+    inputs:
+      azureSubscription: PSResourceGetACR
+      azurePowerShellVersion: LatestVersion
+      ScriptType: InlineScript
+      inline: |
+        $modulePath = Join-Path -Path $env:AGENT_TEMPDIRECTORY -ChildPath 'TempModules'
```
Comment (adityapatwardhan) [inline .ci/test.yml:99]: Maybe just run ACR tests here

File: test/FindPSResourceTests/FindPSResourceACRServer.Tests.ps1 [hunk 2]
```diff
@@ -12,8 +12,21 @@ Describe 'Test HTTP Find-PSResource for ACR Server Protocol' -tags 'CI' {
         $ACRRepoName = "ACRRepo"
         $ACRRepoUri = "https://psresourcegettest.azurecr.io"
         Get-NewPSResourceRepositoryFile
-        $psCredInfo = New-Object Microsoft.PowerShell.PSResourceGet.UtilClasses.PSCredentialInfo ("SecretStore", "$env:TENANTID")
-        Register-PSResourceRepository -Name $ACRRepoName -ApiVersion 'acr' -Uri $ACRRepoUri -CredentialInfo $psCredInfo -Verbose
+
+        Write-Verbose -Verbose "UsingAzAuth = $env:USINGAZAUTH"
```
Comment (adityapatwardhan) [inline test/FindPSResourceTests/FindPSResourceACRServer.Tests.ps1]: Remove this

**Review by anamnavi — APPROVED** [review]
LGTM

File: test/FindPSResourceTests/FindPSResourceACRServer.Tests.ps1 (hunk 2 above)
Comment (adityapatwardhan) [reply test/FindPSResourceTests/FindPSResourceACRServer.Tests.ps1]: Removed

File: .ci/test.yml (hunk 1 above)
Comment (adityapatwardhan) [reply .ci/test.yml:99]: Updated

**Pull request merged**